src/signframework/sign_sha256/sign_sha256.c

```diff
@@ -0,0 +1,630 @@
+            File_Printf(projectLogFile, messageFile,
+                        "ERROR1019: attachment length %u not %u\n",
+                        digestLength, SHA256_SIZE);
+            rc = ERROR_CODE;
+        }
+    }
+    if (rc == 0) {
+        if (verbose) fprintf(messageFile, "Sign: Prepending OID\n");
+        /* prepend OID */
+        memcpy(hash256, sha256_rsa_oid, sizeof(sha256_rsa_oid));
+        /* append digest */
+        memcpy(hash256 + sizeof(sha256_rsa_oid), digest, SHA256_SIZE);
+    }
+    /* sign with the coprocessor.  The coprocessor doesn't know the digest algorithm.  It just
+       signs an OID + digest1 */
+    if (rc == 0) {
+        if (verbose) PrintAll(messageFile,
+                              "Sign: hash to sign",
+                              sizeof(sha256_rsa_oid) + SHA256_SIZE,
+                              hash256);
+        signatureLength = MOD_SIZE;
+        rc = Digital_Signature_Generate(&signatureLength,		/* i/o */
+                                        &signatureBitLength,		/* output */
+                                        signature,			/* output */
+                                        keyTokenLength,			/* input */
+                                        keyToken,			/* input */
+                                        sizeof(sha256_rsa_oid) + SHA256_SIZE, /* input */
+                                        hash256,            /* input */
+                                        SIGN_PKCS_1_1);     /* input */
+
+    }
+    /* create the audit log entry */
+    if (rc == 0) {
+        if (verbose) fprintf(messageFile, "Sign: Updating audit log\n");
+        /* binary data as printable */
+        char pubkey_string[N_SIZE_MAX * 4];
+        char digest_string[SHA256_SIZE * 4];
+        char sig_string[N_SIZE_MAX * 4];
+
+        /* get the user and group structures */
+        /* binary to printable */
+        sprintAll(pubkey_string, MOD_SIZE, rsaKeyTokenPublic.n);
+        sprintAll(digest_string, SHA256_SIZE, digest);
+        sprintAll(sig_string, MOD_SIZE, signature);
+        /* send to audit log */
+        fprintf(projectLogFile, "\tPublic Key:\n %s\n", pubkey_string);
+        fprintf(projectLogFile, "\tDigest:\n %s\n", digest_string);
+        fprintf(projectLogFile, "\tSignature:\n %s\n", sig_string);
+    }
+    /*
+      The verify functions should never fail.  They are just sanity checks on the code.
+    */
+    /* sanity check on the signature length */
+    if (rc == 0) {
+        if (signatureLength != MOD_SIZE) {
+            File_Printf(projectLogFile, messageFile,
+                        "ERROR1001: signature invalid length %lu\n", signatureLength);
+            rc = ERROR_CODE;
+        }
+    }
+    /* verify the signature with the coprocessor key CCA token */
+    if (rc == 0) {
+        if (verbose) fprintf(messageFile,
+                             "Sign: verify signature with the coprocessor key token\n");
+        enum SignMode signMode = SIGN_PKCS_1_1;
```

Comment:
inconsistency: in the previous function on line 276 you passed in the SignMode directly. Now you are defining the variable. I wouldn't be opposed to just defining it at the top of the function (since I believe it is a parameter in 512).